config/config.go

```diff
@@ -140,7 +144,11 @@ func (l *Loader) getResolver() FileResolver {
 // YamlProvider returns function to create Yaml based configuration provider
 func (l *Loader) YamlProvider() ProviderFunc {
 	return func() (Provider, error) {
-		return NewYAMLProviderFromFiles(false, l.getResolver(), l.getFiles()...), nil
+		static := NewYAMLProviderFromFiles(false, l.getResolver(), l.getStaticFiles()...)
+		interpolated := NewYAMLProviderWithExpand(false, l.getResolver(), os.LookupEnv, l.getFiles()...)
```

Comment:
nit: prefer consistent words i.e. either interpolate or expand. Rename to NewYAMLProviderWithInterpolate.